### PR TITLE
Improve FMW Infrastructure Image

### PR DESCRIPTION
**Pull request by mriccell**
1- Remove unsupported version 12.2.1.2
2-Do not auto generate Admin password
3-Supply admin server and database credentials via a domain.properties file
4-Consolidate all env vars into single file

### CODE REVIEW

**Comment by mriccell** [comment]
@markxnelson could you please review?

File: OracleFMWInfrastructure/dockerfiles/12.2.1.3/container-scripts/createInfraDomain.py [hunk 1]
```diff
@@ -49,28 +36,51 @@ class Infra12213Provisioner:
...
         setOption('DomainName', name)
         setOption('JavaHome', self.javaHome)
-        setOption('ServerStartMode', 'prod')
+        setOption('ServerStartMode', prodMode)
         set('Name', domainName)
+
+        # Set Administration Port
+        # =======================
+        if adminPortEnabled != "false":
+         set('AdministrationPort', int(administrationPort))
```
Comment (markxnelson) [inline OracleFMWInfrastructure/dockerfiles/12.2.1.3/container-scripts/createInfraDomain.py]: two spaces?

Comment (markxnelson) [reply OracleFMWInfrastructure/dockerfiles/12.2.1.3/container-scripts/createInfraDomain.py]: not sure if it matters?

File: OracleFMWInfrastructure/dockerfiles/12.2.1.3/container-scripts/createInfraDomain.py [hunk 2]
```diff
@@ -89,18 +101,25 @@ def createBaseDomain(self, name, user, password):
...
-            if server == 'AdminServer':
-                cd('Server/' + server)
-                for param in self.SERVERS[server]:
-                    set(param, self.SERVERS[server][param])
-                continue
-            create(server, 'Server')
-            cd('Server/' + server)
-            for param in self.SERVERS[server]:
-                set(param, self.SERVERS[server][param])
+        # T3 Channel configuration
```
Comment (markxnelson) [inline OracleFMWInfrastructure/dockerfiles/12.2.1.3/container-scripts/createInfraDomain.py]: generally we do not leave commented out code in the repo...

File: OracleFMWInfrastructure/dockerfiles/12.2.1.3/container-scripts/createOrStartInfraDomain.sh [hunk 3]
```diff
@@ -1,13 +1,15 @@
 #!/bin/bash
 #
 #
-# Copyright (c) 2014-2018 Oracle and/or its affiliates. All rights reserved.
```
Comment (markxnelson) [inline OracleFMWInfrastructure/dockerfiles/12.2.1.3/container-scripts/createOrStartInfraDomain.sh]: format wrong

File: OracleFMWInfrastructure/dockerfiles/12.2.1.3/container-scripts/createOrStartInfraDomain.sh [hunk 4]
```diff
@@ -50,86 +38,81 @@ echo "====================================="
...
+   DB_PASS=`awk '{print $1}' $PROPERTIES_FILE | grep db_pass | cut -d "=" -f2`
+   if [ -z "$DB_PASS" ]; then
+      echo "The domain password is blank.  The Admin password must be set in the properties file."
+      exit
+   fi
+   # echo "Database Password $DB_PASS"
+   # Get databasse Schema Password
+   DB_SCHEMA_PASS=`awk '{print $1}' $PROPERTIES_FILE | grep db_schema | cut -d "=" -f2`
+   if [ -z "$DB_SCHEMA_PASS" ]; then
+      echo "The databse schema password is blank.  The cwdatabase schema password must be set in the properties file."
```
Comment (markxnelson) [inline OracleFMWInfrastructure/dockerfiles/12.2.1.3/container-scripts/createOrStartInfraDomain.sh]: is "cwdatabase" a typo?

File: OracleFMWInfrastructure/dockerfiles/12.2.1.3/container-scripts/startManagedServer.sh [hunk 5]
```diff
@@ -1,68 +1,77 @@
 #!/bin/bash
-
-#
-# Copyright (c) 2014-2018 Oracle and/or its affiliates. All rights reserved.
 #
-#Licensed under the Universal Permissive License v 1.0 as shown at http://oss.oracle.com/licenses/upl.
+# Copyright (c) 2014-2019 Oracle and/or its affiliates. All rights reserved.
```
Comment (markxnelson) [inline OracleFMWInfrastructure/dockerfiles/12.2.1.3/container-scripts/startManagedServer.sh]: format wrong

Comment (markxnelson) [reply OracleFMWInfrastructure/dockerfiles/12.2.1.3/container-scripts/startManagedServer.sh]: need to keep UPL statement

File: OracleFMWInfrastructure/dockerfiles/12.2.1.3/properties/domain.properties [hunk 6]
```diff
@@ -0,0 +1,5 @@
+username=mywlsusername
+password=mywlspassword
+db_user=sys
+db_pass=Oradoc_db1
+db_schema=Oradoc_db1
```
Comment (markxnelson) [inline OracleFMWInfrastructure/dockerfiles/12.2.1.3/properties/domain.properties]: hmm, there are normally a number of schemas.... usually we use a PREFIX and then add on to the end what is needed... not sure how many INFRA alone creates, have to check in RCU

File: OracleFMWInfrastructure/README.md [hunk 7]
```diff
@@ -105,43 +88,58 @@ The database is created with the default password `Oradoc_db1`. To change the da
...
+      * `DOMAIN_NAME`                 (default: `infra_domain`)
+      * `ADMINISTRATION_PORT_ENABLED` (default: `true`)
+      * `ADMINISTRATION_PORT`         (default: `9002`)
+      * `MANAGED_NAME`                (default: `infraServer1`)
+      * `MANAGEDSERVER_PORT`          (default: `8001`)
+      * `RCUPREFIX`                   (default: `INFRA01`)
+      * `PRODUCTION_MODE`             (default: `prod`)
 
-  6. Call `docker run` from the `dockerfiles/12.2.1.x` directory where the `infraDomain.env.list` file is located and pass in the file name at runtime.
+**NOTE**: For security, the Administration port 9002 is enabled by default, before running the container in FMW Infrastructure  12.2.1.3. An alternative is to not enable Administration port when you issue the docker run command, set `ADMINISTRTATION_PORT_ENABLED` to false. If you intend to run these images in production, then you must change the Production Mode to `production`. When you set the `DOMAIN_NAME`, the `DOMAIN_HOME=/u01/oracle/user_projects/domains/$DOMAIN_NAME`.
```
[the remark this answers is not in the document]
Comment (mriccell) [reply OracleFMWInfrastructure/README.md]: enabling  ADMINISTRTATION_PORT_ENABLED require all communication to go via the adminport.  For development users do not need this added security and can disable Admin port and just use listen port.

File: OracleFMWInfrastructure/dockerfiles/12.2.1.3/container-scripts/createInfraDomain.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mriccell) [reply OracleFMWInfrastructure/dockerfiles/12.2.1.3/container-scripts/createInfraDomain.py]: I do not understand two spaces?

File: OracleFMWInfrastructure/dockerfiles/12.2.1.3/properties/domain.properties (hunk 6 above)
Comment (mriccell) [reply OracleFMWInfrastructure/dockerfiles/12.2.1.3/properties/domain.properties]: these are properties only for the db in a container.  The RCU properties like the PREFIX are under properties/setEnv.txt

File: OracleFMWInfrastructure/samples/12213-domain-in-volume/README.md [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+
+        `$ sh run_admin_server.sh`
+
+**NOTE**: To have access to the `RCU.out` map volume `/u01/oracle/` in the admin server container. 
+
+  To run Managed Server with base name `infraMS` pass in to the scrtipt `run_managed_server.sh` the name of the managed server you want to run and the host port that will be mapped to the managed server port 8001. To run managed server one with name 'infraMS1` and mapped to host port 98001 call:
+
+        `$ sh run_managed_server.sh infraMS1 98001`
+
+ To run managed server two with name 'infraMS2` and mapped to host port 98002 call:
```
[the remark this answers is not in the document]
Comment (mriccell) [reply OracleFMWInfrastructure/samples/12213-domain-in-volume/README.md]: fixed

File: OracleFMWInfrastructure/samples/12213-domain-in-volume/README.md [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+
+  Start a container to launch the Administration and Managed Servers from the image created in step 1.
+
+  To run an Administration Server container, call:
+
+        `$ sh run_admin_server.sh`
+
+**NOTE**: To have access to the `RCU.out` map volume `/u01/oracle/` in the admin server container. 
+
+  To run Managed Server with base name `infraMS` pass in to the scrtipt `run_managed_server.sh` the name of the managed server you want to run and the host port that will be mapped to the managed server port 8001. To run managed server one with name 'infraMS1` and mapped to host port 98001 call:
```
[the remark this answers is not in the document]
Comment (mriccell) [reply OracleFMWInfrastructure/samples/12213-domain-in-volume/README.md]: fixed

**Comment by Djelibeybi** [comment]
@mriccell when you feel this is complete and ready to be merged, please remove the "WIP" from the title of the pull request and remove the "WIP - DO NOT MERGE" label.

**Comment by mriccell** [comment]
@Djelibeybi these changes are ready to be merged, do you mind merging?  Mark has reviewed but I guess is not in the admin list.

**Pull request merged**